darts/models/forecasting/transformer_model.py

```diff
@@ -22,6 +22,20 @@
 FFN = GLU_FFN + BUILT_IN
 
 
+class _CustomFeedForwardEncoderLayer(nn.TransformerEncoderLayer):
+    # overwrite the feed forward block
+    def _ff_block(self, x):
+        x = self.activation(x)
+        return self.dropout2(x)
```

Comment:
The glu variants have dropout built-in

Reply:
Yes, but I think we should add the second dropout following the original implementation (see [here](https://github.com/pytorch/pytorch/blob/a61c96492b042947467bb603244301400f0e8d1a/torch/nn/modules/transformer.py#L536)):

```
    # feed forward block
    def _ff_block(self, x: Tensor) -> Tensor:
        x = self.linear2(self.dropout(self.activation(self.linear1(x))))
        return self.dropout2(x)
```

then per TransformerEncoderLayer:
```
    def forward(...):
        x = self.norm1(x + self._sa_block(x, src_mask, src_key_padding_mask))
        x = self.norm2(x + self._ff_block(x))
    return x
```

This is equivalent to what they do in the [Annotated Transfomer](https://nlp.seas.harvard.edu/2018/04/03/attention.html#encoder-and-decoder-stacks) with SublayerConnection, EncoderLayer and Position-wise Feed-Forward Networks.

Our current FeedForward class does (changed a bit so it's easier to compare):
```
    def forward(self, x)
        x = self.linear2(self.dropout(self.activation(self.linear1(x))))
        return x
```
So when we overwrite the PyTorch TransformerEncoderLayer, the last dropout would be missing.